kernel/work_q.c

```diff
@@ -92,7 +96,7 @@ int k_delayed_work_submit_to_queue(struct k_work_q *work_q,
 	/* Cancel if work has been submitted */
 	if (work->work_q == work_q) {
 		err = work_cancel(work);
-		if (err < 0) {
+		if (err < 0 && err != -EALREADY) {
```

Comment:
Hmm, I might as well add a comment why we need to handle -EALREADY here since that is likely a resubmit case where the caller is just reusing the same work otherwise we would have to use k_work_init after every work is completed.